src/app/shared/directives/buttons-styling.directive.ts

```diff
@@ -0,0 +1,20 @@
+import { Directive, ElementRef } from '@angular/core';
+
+export type ButtonType = 'primary' | 'default' | 'danger' | null;
+
+@Directive({
+  selector: '[appButtonsStyling]',
+})
+export class ButtonsStylingDirective {
+  constructor(private el: ElementRef) {
+    let btnType = this.el.nativeElement.attributes.buttontype.nodeValue;
+    if (btnType === 'primary') {
+      this.el.nativeElement.style.backgroundColor = '#7ad2f4';
+    } else if (btnType === 'default') {
+      this.el.nativeElement.style.backgroundColor = 'transparent';
+      this.el.nativeElement.style.color = '#545454';
+    } else if (btnType === 'danger') {
+      this.el.nativeElement.style.backgroundColor = 'red';
+    }
+  }
```

Comment:
Just a little code style refactor: it's better to write a switch in this situation, cause, for example, you may one day want to add 'warning' buttonType and there would be another _else if_ 
 Also you can define a method inside your class, which set colors, and call it in switch

```
constructor(private el: ElementRef) {
    let btnType = this.el.nativeElement.attributes.buttontype.nodeValue;
    switch (btnType) {
      case 'primary': this.setColors('#7ad2f4'); break;
      case 'danger': this.setColors('red'); break;
      case 'default':
      default: this.setColors('transparent', '#545454');
    }
  }

  setColors(bkgColor: string, color: string = 'initial') {
    this.el.nativeElement.style.backgroundColor = bkgColor;
    this.el.nativeElement.style.color = color;
  }
```


Or you can define a Map, where a key is your type, value is an object with needed colors. Then you can just get needed values by key and assign it

Reply:
OK! Thanks!